design-system/packages/segmented-control/src/SegmentedControl.tsx

```diff
@@ -89,12 +90,18 @@ export const SegmentedControl = ({
   useEffect(() => {
     if (animate && rootRef.current instanceof HTMLElement) {
       let nodes = Array.from(rootRef.current.children);
-      let selected = nodes[selectedIndex];
+      let selected = selectedIndex !== undefined && nodes[selectedIndex];
```

Comment:
now that -1 is a potential value for selectedIndex, ensure we account for when selected is `undefined`